### PR TITLE
[train] add FAQ

### LINKED ISSUE #22081: Ray Train / Ray Tune fails with Matplotlib
This simple script finishes training, but then the actor coordinating the training dies at the end, causing the fault tolerance engine in Ray to re-run, creating an endless loop.

It may be caused by the inclusion of matplotlib, but that doesn't matter -- data scientists will often have such code, and Ray Train shouldn't cause it to explode.

## System info

```
% which python
/Users/will/opt/anaconda3/envs/torch_to_train/bin/python
% which ipython
/Users/will/opt/anaconda3/envs/torch_to_train/bin/ipython
% python --version
Python 3.8.12
% ipython
In [1]: import ray
In [2]: ray.__version__
Out[2]: '1.9.2'
```

## Code:

```python
# -*- coding: utf-8 -*-
import torch
import torchvision
import torchvision.transforms as transforms
import matplotlib.pyplot as plt
import numpy as np
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim
 
from ray import train
import ray.train.torch
from ray.train import Trainer
from ray.train.torch import TorchConfig
 
 
########################################################################
# The output of torchvision datasets are PILImage images of range [0, 1].
# We transform them to Tensors of normalized range [-1, 1].
 
 
########################################################################
# Let us show some of the training images, for fun.
 
 
def imshow(img):
   img = img / 2 + 0.5     # unnormalize
   npimg = img.numpy()
   plt.imshow(np.transpose(npimg, (1, 2, 0)))
   plt.show()
 
 
 
class Net(nn.Module):
   def __init__(self):
       super().__init__()
       self.conv1 = nn.Conv2d(3, 6, 5)
       self.pool = nn.MaxPool2d(2, 2)
       self.conv2 = nn.Conv2d(6, 16, 5)
       self.fc1 = nn.Linear(16 * 5 * 5, 120)
       self.fc2 = nn.Linear(120, 84)
       self.fc3 = nn.Linear(84, 10)
 
   def forward(self, x):
       x = self.pool(F.relu(self.conv1(x)))
       x = self.pool(F.relu(self.conv2(x)))
       x = torch.flatten(x, 1) # flatten all dimensions except batch
       x = F.relu(self.fc1(x))
       x = F.relu(self.fc2(x))
       x = self.fc3(x)
       return x
 
 
def my_train_func(config):
   # setup data and transofmrers
   transform = transforms.Compose(
       [transforms.ToTensor(),
       transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))]
   )
 
   batch_size = 4
 
   trainset = torchvision.datasets.CIFAR10(root='./data', train=True,
                                           download=True, transform=transform)
   trainloader = torch.utils.data.DataLoader(trainset, batch_size=batch_size,
                                           shuffle=True, num_workers=2)
 
   testset = torchvision.datasets.CIFAR10(root='./data', train=False,
                                       download=True, transform=transform)
   testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size,
                                           shuffle=False, num_workers=2)
 
   classes = ('plane', 'car', 'bird', 'cat',
           'deer', 'dog', 'frog', 'horse', 'ship', 'truck')
  
   # setup model
   net = Net()
   net = train.torch.prepare_model(net)
 
   # setup loss/optimizer
   criterion = nn.CrossEntropyLoss()
   optimizer = optim.SGD(net.parameters(), lr=0.001, momentum=0.9)
 
       # get some random training images
   dataiter = iter(trainloader)
   images, labels = dataiter.next()
   # # show images
   # imshow(torchvision.utils.make_grid(images))
   # # print labels
   # print(' '.join(f'{classes[labels[j]]:5s}' for j in range(batch_size)))
 
   for epoch in range(2):  # loop over the dataset multiple times
 
       running_loss = 0.0
       for i, data in enumerate(trainloader, 0):
           # get the inputs; data is a list of [inputs, labels]
           inputs, labels = data
 
           # zero the parameter gradients
           optimizer.zero_grad()
 
           # forward + backward + optimize
           outputs = net(inputs)
           loss = criterion(outputs, labels)
           loss.backward()
           optimizer.step()
 
           # print statistics
           running_loss += loss.item()
           if i % 2000 == 1999:    # print every 2000 mini-batches
               print(f'[{epoch + 1}, {i + 1:5d}] loss: {running_loss / 2000:.3f}')
               running_loss = 0.0
 
   print('Finished Training')
 
   ########################################################################
   # Let's quickly save our trained model:
 
   PATH = './cifar_net.pth'
   torch.save(net.state_dict(), PATH)
 
   ########################################################################
   # See `here <https://pytorch.org/docs/stable/notes/serialization.html>`_
   # for more details on saving PyTorch models.
   #
   # 5. Test the network on the test data
   # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   #
   # We have trained the network for 2 passes over the training dataset.
   # But we need to check if the network has learnt anything at all.
   #
   # We will check this by predicting the class label that the neural network
   # outputs, and checking it against the ground-truth. If the prediction is
   # correct, we add the sample to the list of correct predictions.
   #
   # Okay, first step. Let us display an image from the test set to get familiar.
 
   dataiter = iter(testloader)
   images, labels = dataiter.next()
 
   # print images
   imshow(torchvision.utils.make_grid(images))
   print('GroundTruth: ', ' '.join(f'{classes[labels[j]]:5s}' for j in range(4)))
 
   ########################################################################
   # Next, let's load back in our saved model (note: saving and re-loading the model
   # wasn't necessary here, we only did it to illustrate how to do so):
 
   net = Net()
   net.load_state_dict(torch.load(PATH))
 
   ########################################################################
   # Okay, now let us see what the neural network thinks these examples above are:
 
   outputs = net(images)
 
   ########################################################################
   # The outputs are energies for the 10 classes.
   # The higher the energy for a class, the more the network
   # thinks that the image is of the particular class.
   # So, let's get the index of the highest energy:
   _, predicted = torch.max(outputs, 1)
 
   print('Predicted: ', ' '.join(f'{classes[predicted[j]]:5s}'
                               for j in range(4)))
 
   ########################################################################
   # The results seem pretty good.
   #
   # Let us look at how the network performs on the whole dataset.
 
   correct = 0
   total = 0
   # since we're not training, we don't need to calculate the gradients for our outputs
   with torch.no_grad():
       for data in testloader:
           images, labels = data
           # calculate outputs by running images through the network
           outputs = net(images)
           # the class with the highest energy is what we choose as prediction
           _, predicted = torch.max(outputs.data, 1)
           total += labels.size(0)
           correct += (predicted == labels).sum().item()
 
   print(f'Accuracy of the network on the 10000 test images: {100 * correct // total} %')
 
   ########################################################################
   # That looks way better than chance, which is 10% accuracy (randomly picking
   # a class out of 10 classes).
   # Seems like the network learnt something.
   #
   # Hmmm, what are the classes that performed well, and the classes that did
   # not perform well:
 
   # prepare to count predictions for each class
   correct_pred = {classname: 0 for classname in classes}
   total_pred = {classname: 0 for classname in classes}
 
   # again no gradients needed
   with torch.no_grad():
       for data in testloader:
           images, labels = data
           outputs = net(images)
           _, predictions = torch.max(outputs, 1)
           # collect the correct predictions for each class
           for label, prediction in zip(labels, predictions):
               if label == prediction:
                   correct_pred[classes[label]] += 1
               total_pred[classes[label]] += 1
 
 
   # print accuracy for each class
   for classname, correct_count in correct_pred.items():
       accuracy = 100 * float(correct_count) / total_pred[classname]
       print(f'Accuracy for class: {classname:5s} is {accuracy:.1f} %')
 
 
 
if __name__ == '__main__':
   config = {}
   trainer = Trainer(backend="torch", num_workers=4)
  
   trainer.start() # set up resources
   trainer.run(my_train_func, config=config)
   trainer.shutdown()
```

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

Adding a FAQ page. Currently has some basic questions that have come up in the past.

Explaining how to use Matplotlib due to threading in the distributed training function.

## Related issue number

Closes #22081

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/user_guide.rst
Comment (matthewdeng) [reply doc/source/train/user_guide.rst:1244]: Yeah, certainly. I had originally decided against creating a separate page in this PR since I wasn't sure how the reader would react if there was just one question listed.

Are there any sections in the existing Train User Guide that you feel should be converted to FAQ? Or any other questions you think are worth answering to start it off?

**Review by amogkam — COMMENTED** [review]
Lgtm, but will defer to @maxpumperla and @worldveil for final approval! Let me know when it's ready to merge!

**Comment by amogkam** [comment]
@matthewdeng do you think this is ready to merge? Should we get another review from @maxpumperla or @worldveil?

**Review by amogkam — APPROVED** [review]

**Pull request merged**
